React/Views/RCTView.m

```diff
@@ -890,6 +890,13 @@ - (void)reactSetFrame:(CGRect)frame
   }
 }
 
+#if TARGET_OS_OSX // [TODO(macOS https://github.com/microsoft/react-native-macos/issues/399)
+- (void)setClipsToBounds:(BOOL)clipsToBounds {
+  [super setClipsToBounds:clipsToBounds];
+  self.layer.masksToBounds = clipsToBounds;
```

Comment:
This is explicitly documented as undefined behavior. See https://developer.apple.com/library/archive/releasenotes/AppKit/RN-AppKit/#10_13Layer-backed%20Views.
`macOS 10.13 is more consistent around updates to properties of views' layers (that is, layers that are the .layer of some NSView, however created). As before, if an application modifies such a layer by changing any of the following properties, the behavior of the application is undefined: bounds, position, zPosition, anchorPoint, anchorPointZ, transform, affineTransform, frame, hidden, geometryFlipped, masksToBounds, opaque, compositingFilter, filters, shadowColor, shadowOpacity, shadowOffset, shadowRadius, shadowPath, layoutManager.
`
I'm not super clear on what the overflow:'hidden' behavior is supposed to do and why it's valuable, but we need to come up with another approach

Reply:
Perhaps we could set the view's layer to a custom layer.  I tried the following locally and it seemed to work:
 
in `RCTView.m` I created a 
```ObjC
@interface RCTNoClippingLayer: CALayer
@end
@implementation RCTNoClippingLayer
-(BOOL) masksToBounds {
  return NO;
}
@end
```
and then added a `setClipsToBounds:` override that does
```objc
- (void)setClipsToBounds:(BOOL)clipsToBounds {
  [super setClipsToBounds:clipsToBounds];
  if (clipsToBounds) {
    self.layer = [RCTNoClippingLayer new];
  }
}
```
In RNTester it works.   The impl is probably too simple though.   I guess we'd have to cache the original `CALayer` and restore it if the prop is set to `NO`.  And the impl should probably be put up a level on `RCTUIView` since that's where `clipsToBounds` is defined.    But I think in principal it avoids the undefined behavior as documented by Apple of setting props on existing layers.